use TrackRef instead of TrackId

uklotzde opened the pull request:
Replace `QList<TrackId>` with `QList<TrackRef>` as return type when purging tracks.

The temporary copy in `TrackCollection` is compensated by avoiding an unnecessary database access in `TrackCollectionManager`.

See also: https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/Export.20to.20Engine.20Prime/near/188593776

mr-smidge committed:
Use TrackRef rather than TrackId

uklotzde committed:
TrackDAO: Replace getAllTrackIds() with getAllTrackRefs()

uklotzde committed:
Rename function

mr-smidge commented on src/library/dao/trackdao.h:
```diff
@@ -53,16 +53,15 @@ class TrackDAO : public QObject, public virtual DAO, public virtual GlobalTrackC
-    QSet<QString> getTrackLocations();
+    QSet<QString> getAllTrackLocations();
     QString getTrackLocation(TrackId trackId);
-    QStringList getTrackLocations(const QList<TrackId>& trackIds);
```
Agree with removing this method - it seems out of place to have an isolated list of locations for specific tracks without knowing the mapping between them!

uklotzde commented:
Ping at anyone. Merge?

daschuer commented:
The code looks good, a manual test is pending.

Be-ing commented:
I got confused and thought this PR was implementing a new feature for exporting to Engine Prime, so I changed the PR title to clarify.

uklotzde commented:
Agreed. I didn't try to find a good title for this PR.

My motivation was to check if the changes required for the Engine Prime export make sense in general. It turned out they do. We should adopt them to avoid merge conflicts and to simplify the development for this feature.

Holzhaus commented:
Compiled fine. I wasn't sure how to test this. I marked a track as hidden, then went to the hidden tracks table and removed it there. This worked fine.

Probably unrelated: I also removed a file for a track. I even tried to load it into a deck (which didn't work obviously). Shouldn't the track be moved from "All tracks" to "Missing tracks" automatically?

uklotzde commented:
> Compiled fine. I wasn't sure how to test this. I marked a track as hidden, then went to the hidden tracks table and removed it there. This worked fine.
> 
> Probably unrelated: I also removed a file for a track. I even tried to load it into a deck (which didn't work obviously). Shouldn't the track be moved from "All tracks" to "Missing tracks" automatically?

Existing behavior should not be affected by modifying the internal database operations. This is out of the scope of this PR.

I have retested with 2.2.3, just to be sure: Tracks that have been deleted on the file system and that cannot be opened are **not** automatically marked as missing. A library rescan is required to achieve this.

uklotzde commented:
Can we merge this? I want to get these finished PRs out of my repo and out of my mind.

Holzhaus commented:
Yup, LGTM.

daschuer commented:
@mr-smidge 
I am currently setting up the contributor list inside the Mixxx about box. 
If it is OK, I like to add your full name as well. Is that OK? 
I have noticed that you have not signed our contributor agreement. 
Can you sign this form now, post merge? 
 https://docs.google.com/a/mixxx.org/spreadsheet/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ
Please comment here when done.

mr-smidge commented:
> @mr-smidge
> I am currently setting up the contributor list inside the Mixxx about box.
> If it is OK, I like to add your full name as well. Is that OK?
> I have noticed that you have not signed our contributor agreement.
> Can you sign this form now, post merge?
> https://docs.google.com/a/mixxx.org/spreadsheet/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ
> Please comment here when done.

Hi @daschuer, I'm happy for you to add my full name to the list.  I've now signed the form.

daschuer commented:
Done.
